### PR TITLE
Add additional UTF-8 validation

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Add UTF-8 validation to namespace
* Add UTF-8 validation to workflow ID

**Why?**
Necessary input validation

**How did you test it?**
N/A

**Potential risks**
N/A

**Is hotfix candidate?**
No

### CODE REVIEW

File: service/frontend/workflowHandler.go [hunk 1]
```diff
@@ -3399,3 +3392,42 @@ func (wh *WorkflowHandler) makeFakeContinuedAsNewEvent(
 		},
 	}, nil
 }
+
+func (wh *WorkflowHandler) validateNamespace(
```
Comment (yiminc) [inline service/frontend/workflowHandler.go:3396]: we don't need to do namespace check. Namespace is validated by NamespaceValidatorInterceptor

File: service/frontend/workflowHandler.go [hunk 2]
```diff
@@ -3399,3 +3392,42 @@ func (wh *WorkflowHandler) makeFakeContinuedAsNewEvent(
...
+	if err := wh.validateUTF8String(str); err != nil {
+		return err
+	}
+	if len(str) > wh.config.MaxIDLengthLimit() {
+		return errWorkflowIDTooLong
+	}
+	return nil
+}
+
+func (wh *WorkflowHandler) validateUTF8String(
```
Comment (yiminc) [inline service/frontend/workflowHandler.go:3426]: can you have a unit test just to verify that \ufffd would be caught by this check.

File: service/frontend/workflowHandler.go (hunk 1 above)
Comment (yiminc) [reply service/frontend/workflowHandler.go:3396]: ignore this, we do need this validation for register namespace.

**Review by yiminc — APPROVED** [review]

File: service/frontend/workflowHandler.go (hunk 2 above)
Comment (wxing1292) [reply service/frontend/workflowHandler.go:3426]: sample

```go
hexStr := "ff"
hexBytes, err := hex.DecodeString(hexStr)
s.NoError(err)
s.False(utf8.ValidString(string(hexBytes)))

fmt.Println("#######")
    prettyPrint := func(input interface{}) string {
    binary, _ := json.MarshalIndent(input, "", "  ")
    return string(binary)
}
fmt.Printf("%v\n", prettyPrint(string(hexBytes)))
fmt.Println("#######")
```

output 
```
#######
"\ufffd"
#######
```

**Pull request merged**